### PR TITLE
Activeadmin v3 support

### LINKED ISSUE #475: ActiveAdmin version 3.0.0 released!
Hey! Just letting you guys know that ActiveAdmin had a new major version recently released with a couple of breaking changes:
https://github.com/activeadmin/activeadmin/releases/tag/v3.0.0

I did a fast search through the project and it seems like it shouldn't really break much on AA_addons. I know you're working in a v2 version of this, so thought you might find it useful to know.

### PR DESCRIPTION
**Pull request by difernandez**
### Motivation / Background

This Pull Request has been created because ActiveAdmin [released v3](https://github.com/activeadmin/activeadmin/releases/tag/v3.0.0) 🎉 . It has some breaking changes, including [requiring of ransack v4](https://github.com/activerecord-hackery/ransack/releases/tag/v4.0.0), so some changes needed to be done on our part to support it.

### Detail

This Pull Request applies a couple of backwards compatible changes that allow this gem to work well with AA v3, and for tests to pass using that version:
- Adds `ransackable_attributes` and `ransackable_associations` to dummy app models, as it is now [explicitly required by ransack v4](https://github.com/activerecord-hackery/ransack/pull/1400)
- Removes some filter predicates (pretty much just `contains`), that [are now removed from AA](https://github.com/activeadmin/activeadmin/pull/8010). These predicates were backports in AA, and they have been replaced by their already existent Ransack equivalents
- Updates activeadmin version in Gemfile, to test and develop using v3

### Additional information

Closes #475 

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a concise description of what changed and why.
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] Documentation has been added or updated if you add a feature or modify an existing one.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature (under the "Unreleased" heading if this is not a version change).
* [x] My changes don't introduce any linter rule violations.

### CODE REVIEW

File: spec/dummy/app/models/category.rb [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+
   def shiny_name
     "My shiny #{name}"
   end
+
+  def self.ransackable_attributes(auth_object = nil)
+    ["created_at", "description", "id", "name", "updated_at"]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/category.rb:12]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/city.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
 class City < ActiveRecord::Base
   belongs_to :region
+  has_many :departments_cities, dependent: :destroy
+  has_many :departments, through: :departments_cities
+
+  def self.ransackable_attributes(auth_object = nil)
+    ["created_at", "id", "information", "name", "region_id", "updated_at"]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/city.rb:10]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/country.rb [hunk 3]
```diff
@@ -1,2 +1,12 @@
 class Country < ActiveRecord::Base
+  has_many :regions, dependent: :destroy
+  has_many :cities, through: :regions
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/country.rb:5]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_attributes(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/country.rb [hunk 4]
```diff
@@ -1,2 +1,12 @@
...
+    ["created_at", "id", "information", "name", "updated_at"]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/country.rb:9]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/invoice.rb [hunk 5]
```diff
@@ -92,4 +92,18 @@ def self.colors
   def set_default_state
     self.state ||= 'pending'
   end
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/invoice.rb:96]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_attributes(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/invoice.rb [hunk 6]
```diff
@@ -92,4 +92,18 @@ def self.colors
...
+    [
+      "aasm_state", "active", "amount", "attachment_content_type", "attachment_file_name",
+      "attachment_file_size", "attachment_updated_at", "category_id", "city_id", "client_id",
+      "color", "created_at", "description", "id", "legal_date", "number", "paid",
+      "photo_content_type", "photo_file_name", "photo_file_size", "photo_updated_at",
+      "picture_data", "position", "shipping_status", "state", "status", "updated_at",
```
Comment (monkeyci) [inline spec/dummy/app/models/invoice.rb:102]: **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Style/TrailingCommaInArrayLiteral: Avoid comma after the last item of an array.

File: spec/dummy/app/models/invoice.rb [hunk 7]
```diff
@@ -92,4 +92,18 @@ def self.colors
...
+    ]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/invoice.rb:106]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/item.rb [hunk 8]
```diff
@@ -4,4 +4,12 @@ class Item < ActiveRecord::Base
   def full_name
     "##{id} - #{name}"
   end
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/item.rb:8]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_attributes(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/item.rb [hunk 9]
```diff
@@ -4,4 +4,12 @@ class Item < ActiveRecord::Base
...
+    ["created_at", "description", "id", "name", "updated_at"]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/item.rb:12]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/region.rb [hunk 10]
```diff
@@ -1,3 +1,12 @@
 class Region < ActiveRecord::Base
   belongs_to :country
+  has_many :cities, dependent: :destroy
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/region.rb:5]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_attributes(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/region.rb [hunk 11]
```diff
@@ -1,3 +1,12 @@
...
+    ["country_id", "created_at", "id", "information", "name", "updated_at"]
+  end
+
+  def self.ransackable_associations(auth_object = nil)
```
Comment (monkeyci) [inline spec/dummy/app/models/region.rb:9]: ⚠️ **[rubocop]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Correctable] Lint/UnusedMethodArgument: Unused method argument - auth_object. If it's necessary, use _ or _auth_object as an argument name to indicate that it won't be used. If it's unnecessary, remove it. You can also write as ransackable_associations(*) if you want the method to accept any arguments but don't care about them.

File: spec/dummy/app/models/invoice.rb [hunk 12]
```diff
@@ -92,4 +92,18 @@ def self.colors
   def set_default_state
     self.state ||= 'pending'
   end
+
+  def self.ransackable_attributes(_auth_object = nil)
```
Comment (ldlsegovia) [inline spec/dummy/app/models/invoice.rb:96]: I understand that now it is mandatory to do this, right?

Comment (difernandez) [reply spec/dummy/app/models/invoice.rb:96]: [yep](https://github.com/activerecord-hackery/ransack/pull/1400)

**Comment by jgmontoya** [comment]
Can we get a release of this before the whole v2 overhaul?

**Comment by difernandez** [comment]
@jgmontoya we just released v1.10.1 , please let us know how it goes if you try it out

**Pull request merged**